code/common/checkinputs.q

```diff
@@ -0,0 +1,276 @@
+    dict:checkeachparam[dict;0b];
+    dict[`checksperformed]:1b;
+    :dict;
+  };
+
+checkdictionary:{[dict]
+    if[not checkkeytype dict;'`$"Input dictionary must have keys of type 11h"];
+    if[not checkrequiredparams dict;'`$.checkinputs.formatstring["Required parameters missing:{}";.checkinputs.getrequiredparams[]except key dict]];
+    if[not checkparamnames dict;'`$.checkinputs.formatstring["Invalid parameter present:{}. For a list of valid parameters, use the command .checkinputs.getvalidparams[]";key[dict]except .checkinputs.getvalidparams[]]];
+    :dict;
+  };
+
+isdictionary:{[dict]$[99h~type dict;:dict;'`$"Input must be a dictionary"]};
+checkkeytype:{[dict]11h~type key dict};
+checkrequiredparams:{[dict]all .checkinputs.getrequiredparams[]in key dict};
+getrequiredparams:{[]exec parameter from .checkinputs.checkinputsconfig where required}
+checkparamnames:{[dict]all key[dict]in .checkinputs.getvalidparams[]};
+getvalidparams:{[]exec parameter from .checkinputs.checkinputsconfig};
+
+checkunaryfunc:{[dict;parameter]
+    dict:checktype[100h;dict;parameter];
+    if[1<>count (get dict parameter)[1];
+        '`$"Postback argument must be a function that takes in one argument only - the argument can be named anything through function signature but must represent the returned results of all other inputs"];}
+
+
+checkinvalidcombinations:{[dict]
+    parameters:key dict;
+    xinvalidpairs:select parameter,invalidpairs:invalidpairs inter\:parameters from .checkinputs.checkinputsconfig where parameter in parameters;
+    xinvalidpairs:select from xinvalidpairs where 0<>count'[invalidpairs];
+    if[0=count xinvalidpairs;:dict];
+    :checkeachpair[raze each flip xinvalidpairs];
+  };
+
+checkeachpair:{[invalidpair]'`$.checkinputs.formatstring["Parameter:{parameter} cannot be used in conjunction with parameter(s):{invalidpairs}";invalidpair]};
+
+// function to check if any parameters are repeated
+checkrepeatparams:{[dict]
+    if[any repeats:1<count each group key dict;
+        '`$.checkinputs.formatstring["{} parameter(s) used more than once";where repeats]];
+    :dict;};
+
+// loop thorugh input parameters to execute parameter specific checks
+checkeachparam:{[dict;isrequired]
+    config:select from .checkinputs.checkinputsconfig where parameter in key dict,required=isrequired;
+    :checkparam/[dict;config];
+  };
+
+// extract parameter specific function from config to check the input
+checkparam:{[dict;config]
+    (first config[`checkfunction])[dict;first config`parameter]};
+
+// check tablename parameter is of type symbol
+checktable:{[dict;parameter]:checktype[-11h;dict;parameter];};
+
+// check that endtime is of temporal type and that it is greater than or equal to starttime
+checkendtime:{[dict;parameter]
+    dict:checktimetype[dict;parameter];
+    :checktimeorder dict};
+
+// check that inputted value is of valid type: -12 -14 -15h
+checktimetype:{[dict;parameter]:checktype[-12 -14 -15h;dict;parameter];};
```

Comment:
Might be better to have a global parameter of time types so that if KX decided to add/remove its easy to find and add to the list.